### PR TITLE
Stats: apply API changes to prevent reset

### PR DESCRIPTION
**Pull request by leandroalonso**
We had a few instances of users' complaining that their Stats reset or reduced. This PR applies the latest changes deployed on Staging ensuring:

1. That the backend never "reduces" (or reset) any Stat;
2. That we pull back the stats in case we need

## To test

1. First, change the `Build Configuration` to `Staging` and run the app
2. Login to a staging account
3. Ensure this account has some Stats. If it doesn't, [manipulate these lines](https://github.com/Automattic/pocket-casts-ios/blob/trunk/Modules/Server/Sources/PocketCastsServer/Public/StatsManager.swift#L23-L29) and trigger a refresh (just add numbers there)
4. Then, remove all the `updateQueue.sync` block from the `StatsManager` `init` method. This will simulate having negative stats
5. Go to Stats
6. ✅ Reduced Stats should appear but it should update to the correct one
7. Re-run the app
8. Trigger a refresh from Profile
9. Go to Stats
10. ✅ Stats should be correct

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/Server/Sources/PocketCastsServer/Public/StatsManager.swift [hunk 1]
```diff
@@ -29,6 +29,39 @@ public class StatsManager {
         }
     }
 
+    func updateStatsIfNeeded(savedDynamicSpeed: TimeInterval, savedVariableSpeed: TimeInterval, totalListenedTo: TimeInterval, totalSkipped: TimeInterval, savedAutoSkipping: TimeInterval) {
+        let minimumStatsChangeToUpdate = 100
```
Comment (leandroalonso) [inline Modules/Server/Sources/PocketCastsServer/Public/StatsManager.swift]: I'm debating whether this should be `100` or something bigger or smaller.

Maybe if the local stat is just smaller than remote we should update?

File: Modules/Server/Sources/PocketCastsServer/Public/StatsManager.swift [hunk 2]
```diff
@@ -29,6 +29,39 @@ public class StatsManager {
         }
     }
 
+    func updateStatsIfNeeded(savedDynamicSpeed: TimeInterval, savedVariableSpeed: TimeInterval, totalListenedTo: TimeInterval, totalSkipped: TimeInterval, savedAutoSkipping: TimeInterval) {
+        let minimumStatsChangeToUpdate: TimeInterval = 100
+
+        updateQueue.sync {
+            if savedDynamicSpeed - self.savedDynamicSpeed > minimumStatsChangeToUpdate {
```
Comment (SergioEstevao) [inline Modules/Server/Sources/PocketCastsServer/Public/StatsManager.swift:36]: Nit: Do you want to save only if the difference is positive right? No need to use `abs` ?

**Review by SergioEstevao — APPROVED** [review]
Followed the test instructions and the stats recovered to the correct values.
Just left a small nit that you may want to consider, but feel free to ignore if does not make sense for the scenario.
:shipit:

**Comment by leandroalonso** [comment]
@srjakes are the server changes deployed?

**Comment by srjakes** [comment]
> are the server changes deployed?

The server changes have been deployed.

**Pull request merged**